osm2streets-java/src/lib.rs

```diff
@@ -43,16 +43,61 @@ pub extern "system" fn Java_org_osm2streets_StreetNetwork_create(
 }
 
 #[no_mangle]
-pub unsafe extern "system" fn Java_org_osm2streets_StreetNetwork_toGeojsonPlain(
+#[allow(non_snake_case)]
+pub unsafe extern "system" fn Java_org_osm2streets_StreetNetwork_getRoadSurface(
     env: JNIEnv,
-    java_pointer: JObject,
-) -> jstring {
-    let inner_pointer = env.get_field(java_pointer, "pointer", "J").unwrap();
+    j_self: JObject,
+) -> jobject {
+    let inner_pointer = env.get_field(j_self, "pointer", "J").unwrap();
     let streets = &mut *(inner_pointer.j().unwrap() as *mut StreetNetwork);
 
-    let result = streets.inner.to_geojson().unwrap();
-    let output = env.new_string(result).unwrap();
-    output.into_raw()
+    let feature_collection = streets.inner.to_road_surface();
+
+    // Cache the JNI stuff for performance.
```

Comment:
This is actually not too bad to understand! I'm sure it was not painless to write, though

Reply:
It was a bit of back and forth, mostly learning the JNI concepts that I'm sure you learnt getting the original thing building!

Reply:
Aaand already forgotten, like cramming for an exam ;)